bc_obps/service/tests/data_access_service/test_data_access_opted_in_operation_detail_service.py

```diff
@@ -11,13 +13,18 @@ class TestDataAccessOptedInOperationService:
     @staticmethod
     def test_update_opted_in_operation_detail():
         approved_user_operator = baker.make_recipe('utils.approved_user_operator')
-        users_operation = baker.make_recipe(
-            'utils.operation', operator=approved_user_operator.operator, created_by=approved_user_operator.user
+        create_operation_payload = OperationInformationIn(
+            registration_purpose=Operation.Purposes.OPTED_IN_OPERATION,
+            regulated_products=[1, 2],
+            activities=[1],
+            boundary_map=MOCK_DATA_URL,
+            process_flow_diagram=MOCK_DATA_URL,
+            naics_code_id=1,
+            type='Single Facility Operation',
+            name='Test Opted in Operation',
         )
-        # Marking the operation as opted-in
-        purpose_payload = RegistrationPurposeIn(registration_purpose='Opted-in Operation')
-        OperationServiceV2.set_registration_purpose(
-            approved_user_operator.user.user_guid, users_operation.id, purpose_payload
+        users_operation = OperationServiceV2.create_or_update_operation_v2(
```

Comment:
Is this setup/mocking? The old test was also calling different services than the one it was testing, but if possible, it would be good to do all the setup with baker recipes and then only call the opted_in_operation_detail_service in the test

Reply:
excellent observation. I was extremely sloppy in updating this test. Fixed it now.